include/glow/Optimizer/Optimizer.h

```diff
@@ -51,7 +52,14 @@ void lower(Function *F, CompilationContext &cctx, const Backend *B = nullptr,
            const KindSet &doNotLowerKinds = {});
 
 /// Dead code elimination.
-void DCE(Function *F);
+class DCE : public FunctionPass {
+private:
+  /// Implementation of DCE. \returns whether the pass modifies \p F.
+  bool passImpl(Function *F) override;
+
+public:
+  llvm::StringRef getName() const override { return "DCE"; }
```

Comment:
do we have any test which runs through the getName(), I suspect "DCE" might go out of scope and gets destroyed while StringRef still points to it. Maybe have static member passname inside the DCE class and return it here.

Reply:
Thanks for double checking. "String literals have static storage duration, and thus exist in memory for the life of the program", so we should be safe here.